// @pulumi/aiven/accountTeamProject.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountTeamProject = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Account Team Project resource allows the creation and management of an Account Team Project.
 *
 * It is intended to link an existing project to the existing account team.
 * It is important to note that the project should have an `accountId` property set equal to the
 * account team you are trying to link to this project.
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/accountTeamProject:AccountTeamProject account_team_project1 account_id/team_id/project_name
 * ```
 */
class AccountTeamProject extends pulumi.CustomResource {
    /**
     * Get an existing AccountTeamProject resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccountTeamProject(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccountTeamProject.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountTeamProject.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["projectName"] = state ? state.projectName : undefined;
            resourceInputs["teamId"] = state ? state.teamId : undefined;
            resourceInputs["teamType"] = state ? state.teamType : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.teamId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'teamId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["teamId"] = args ? args.teamId : undefined;
            resourceInputs["teamType"] = args ? args.teamType : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccountTeamProject.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccountTeamProject = AccountTeamProject;
/** @internal */
AccountTeamProject.__pulumiType = 'aiven:index/accountTeamProject:AccountTeamProject';
