// @pulumi/aiven/awsPrivatelink.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsPrivatelink = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The AWS Privatelink resource allows the creation and management of Aiven AWS Privatelink for a services.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aiven from "@pulumi/aiven";
 *
 * const foo = new aiven.AwsPrivatelink("foo", {
 *     project: data.aiven_project.foo.project,
 *     serviceName: aiven_kafka.bar.service_name,
 *     principals: ["arn:aws:iam::012345678901:user/mwf"],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/awsPrivatelink:AwsPrivatelink foo project/service_name
 * ```
 */
class AwsPrivatelink extends pulumi.CustomResource {
    /**
     * Get an existing AwsPrivatelink resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AwsPrivatelink(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AwsPrivatelink.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AwsPrivatelink.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["awsServiceId"] = state ? state.awsServiceId : undefined;
            resourceInputs["awsServiceName"] = state ? state.awsServiceName : undefined;
            resourceInputs["principals"] = state ? state.principals : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["serviceName"] = state ? state.serviceName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.principals === undefined) && !opts.urn) {
                throw new Error("Missing required property 'principals'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["principals"] = args ? args.principals : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["awsServiceId"] = undefined /*out*/;
            resourceInputs["awsServiceName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AwsPrivatelink.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AwsPrivatelink = AwsPrivatelink;
/** @internal */
AwsPrivatelink.__pulumiType = 'aiven:index/awsPrivatelink:AwsPrivatelink';
