// @pulumi/aiven/azurePrivatelinkConnectionApproval.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzurePrivatelinkConnectionApproval = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Azure privatelink approve resource waits for an aiven privatelink connection on a service and approves it with associated endpoint IP
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/azurePrivatelinkConnectionApproval:AzurePrivatelinkConnectionApproval approval project/service_name
 * ```
 */
class AzurePrivatelinkConnectionApproval extends pulumi.CustomResource {
    /**
     * Get an existing AzurePrivatelinkConnectionApproval resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AzurePrivatelinkConnectionApproval(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AzurePrivatelinkConnectionApproval.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AzurePrivatelinkConnectionApproval.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["endpointIpAddress"] = state ? state.endpointIpAddress : undefined;
            resourceInputs["privatelinkConnectionId"] = state ? state.privatelinkConnectionId : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["serviceName"] = state ? state.serviceName : undefined;
            resourceInputs["state"] = state ? state.state : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["endpointIpAddress"] = args ? args.endpointIpAddress : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["privatelinkConnectionId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AzurePrivatelinkConnectionApproval.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AzurePrivatelinkConnectionApproval = AzurePrivatelinkConnectionApproval;
/** @internal */
AzurePrivatelinkConnectionApproval.__pulumiType = 'aiven:index/azurePrivatelinkConnectionApproval:AzurePrivatelinkConnectionApproval';
