// @pulumi/aiven/azureVpcPeeringConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureVpcPeeringConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Azure VPC Peering Connection resource allows the creation and management of Aiven VPC Peering Connections.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aiven from "@pulumi/aiven";
 *
 * const foo = new aiven.AzureVpcPeeringConnection("foo", {
 *     vpcId: data.aiven_project_vpc.vpc.id,
 *     azureSubscriptionId: "xxxxxx",
 *     peerResourceGroup: "my-pr1",
 *     vnetName: "my-vnet1",
 *     peerAzureAppId: "xxxxxx",
 *     peerAzureTenantId: "xxxxxx",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/azureVpcPeeringConnection:AzureVpcPeeringConnection foo project_name/vpc_id/azure_subscription_id/vnet_name
 * ```
 */
class AzureVpcPeeringConnection extends pulumi.CustomResource {
    /**
     * Get an existing AzureVpcPeeringConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AzureVpcPeeringConnection(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AzureVpcPeeringConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AzureVpcPeeringConnection.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["azureSubscriptionId"] = state ? state.azureSubscriptionId : undefined;
            resourceInputs["peerAzureAppId"] = state ? state.peerAzureAppId : undefined;
            resourceInputs["peerAzureTenantId"] = state ? state.peerAzureTenantId : undefined;
            resourceInputs["peerResourceGroup"] = state ? state.peerResourceGroup : undefined;
            resourceInputs["peeringConnectionId"] = state ? state.peeringConnectionId : undefined;
            resourceInputs["state"] = state ? state.state : undefined;
            resourceInputs["stateInfo"] = state ? state.stateInfo : undefined;
            resourceInputs["vnetName"] = state ? state.vnetName : undefined;
            resourceInputs["vpcId"] = state ? state.vpcId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.azureSubscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureSubscriptionId'");
            }
            if ((!args || args.peerAzureAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peerAzureAppId'");
            }
            if ((!args || args.peerAzureTenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peerAzureTenantId'");
            }
            if ((!args || args.peerResourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peerResourceGroup'");
            }
            if ((!args || args.vnetName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vnetName'");
            }
            if ((!args || args.vpcId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            resourceInputs["azureSubscriptionId"] = args ? args.azureSubscriptionId : undefined;
            resourceInputs["peerAzureAppId"] = args ? args.peerAzureAppId : undefined;
            resourceInputs["peerAzureTenantId"] = args ? args.peerAzureTenantId : undefined;
            resourceInputs["peerResourceGroup"] = args ? args.peerResourceGroup : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vpcId"] = args ? args.vpcId : undefined;
            resourceInputs["peeringConnectionId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["stateInfo"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AzureVpcPeeringConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AzureVpcPeeringConnection = AzureVpcPeeringConnection;
/** @internal */
AzureVpcPeeringConnection.__pulumiType = 'aiven:index/azureVpcPeeringConnection:AzureVpcPeeringConnection';
