// @pulumi/aiven/openSearchAclConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenSearchAclConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Opensearch resource allows the creation and management of Aiven Opensearch services.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aiven from "@pulumi/aiven";
 *
 * const fooProject = aiven.getProject({
 *     project: "example_project",
 * });
 * const bar = new aiven.OpenSearch("bar", {
 *     project: fooProject.then(fooProject => fooProject.project),
 *     cloudName: "google-europe-west1",
 *     plan: "startup-4",
 *     serviceName: "example_service_name",
 *     maintenanceWindowDow: "monday",
 *     maintenanceWindowTime: "10:00:00",
 * });
 * const fooServiceUser = new aiven.ServiceUser("fooServiceUser", {
 *     serviceName: bar.serviceName,
 *     project: fooProject.then(fooProject => fooProject.project),
 *     username: "user-example",
 * });
 * const fooOpenSearchAclConfig = new aiven.OpenSearchAclConfig("fooOpenSearchAclConfig", {
 *     project: fooProject.then(fooProject => fooProject.project),
 *     serviceName: bar.serviceName,
 *     enabled: true,
 *     extendedAcl: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/openSearchAclConfig:OpenSearchAclConfig foo project/service_name
 * ```
 */
class OpenSearchAclConfig extends pulumi.CustomResource {
    /**
     * Get an existing OpenSearchAclConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OpenSearchAclConfig(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OpenSearchAclConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OpenSearchAclConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["extendedAcl"] = state ? state.extendedAcl : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["serviceName"] = state ? state.serviceName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["extendedAcl"] = args ? args.extendedAcl : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(OpenSearchAclConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OpenSearchAclConfig = OpenSearchAclConfig;
/** @internal */
OpenSearchAclConfig.__pulumiType = 'aiven:index/openSearchAclConfig:OpenSearchAclConfig';
