// @pulumi/aiven/projectVpc.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectVpc = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Project VPC resource allows the creation and management of Aiven Project VPCs.
 *
 * ## Import
 *
 * ```sh
 *  $ pulumi import aiven:index/projectVpc:ProjectVpc myvpc project/id
 * ```
 */
class ProjectVpc extends pulumi.CustomResource {
    /**
     * Get an existing ProjectVpc resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ProjectVpc(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProjectVpc.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProjectVpc.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cloudName"] = state ? state.cloudName : undefined;
            resourceInputs["networkCidr"] = state ? state.networkCidr : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["state"] = state ? state.state : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.cloudName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cloudName'");
            }
            if ((!args || args.networkCidr === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkCidr'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            resourceInputs["cloudName"] = args ? args.cloudName : undefined;
            resourceInputs["networkCidr"] = args ? args.networkCidr : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ProjectVpc.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ProjectVpc = ProjectVpc;
/** @internal */
ProjectVpc.__pulumiType = 'aiven:index/projectVpc:ProjectVpc';
