// @pulumi/aiven/serviceIntegrationEndpoint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceIntegrationEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The Service Integration Endpoint resource allows the creation and management of Aiven Service Integration Endpoints.
 */
class ServiceIntegrationEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing ServiceIntegrationEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceIntegrationEndpoint(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServiceIntegrationEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceIntegrationEndpoint.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["datadogUserConfig"] = state ? state.datadogUserConfig : undefined;
            resourceInputs["endpointConfig"] = state ? state.endpointConfig : undefined;
            resourceInputs["endpointName"] = state ? state.endpointName : undefined;
            resourceInputs["endpointType"] = state ? state.endpointType : undefined;
            resourceInputs["externalAwsCloudwatchLogsUserConfig"] = state ? state.externalAwsCloudwatchLogsUserConfig : undefined;
            resourceInputs["externalAwsCloudwatchMetricsUserConfig"] = state ? state.externalAwsCloudwatchMetricsUserConfig : undefined;
            resourceInputs["externalElasticsearchLogsUserConfig"] = state ? state.externalElasticsearchLogsUserConfig : undefined;
            resourceInputs["externalGoogleCloudLoggingUserConfig"] = state ? state.externalGoogleCloudLoggingUserConfig : undefined;
            resourceInputs["externalKafkaUserConfig"] = state ? state.externalKafkaUserConfig : undefined;
            resourceInputs["externalOpensearchLogsUserConfig"] = state ? state.externalOpensearchLogsUserConfig : undefined;
            resourceInputs["externalSchemaRegistryUserConfig"] = state ? state.externalSchemaRegistryUserConfig : undefined;
            resourceInputs["jolokiaUserConfig"] = state ? state.jolokiaUserConfig : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["prometheusUserConfig"] = state ? state.prometheusUserConfig : undefined;
            resourceInputs["rsyslogUserConfig"] = state ? state.rsyslogUserConfig : undefined;
            resourceInputs["signalfxUserConfig"] = state ? state.signalfxUserConfig : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.endpointName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointName'");
            }
            if ((!args || args.endpointType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointType'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            resourceInputs["datadogUserConfig"] = args ? args.datadogUserConfig : undefined;
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["endpointType"] = args ? args.endpointType : undefined;
            resourceInputs["externalAwsCloudwatchLogsUserConfig"] = args ? args.externalAwsCloudwatchLogsUserConfig : undefined;
            resourceInputs["externalAwsCloudwatchMetricsUserConfig"] = args ? args.externalAwsCloudwatchMetricsUserConfig : undefined;
            resourceInputs["externalElasticsearchLogsUserConfig"] = args ? args.externalElasticsearchLogsUserConfig : undefined;
            resourceInputs["externalGoogleCloudLoggingUserConfig"] = args ? args.externalGoogleCloudLoggingUserConfig : undefined;
            resourceInputs["externalKafkaUserConfig"] = args ? args.externalKafkaUserConfig : undefined;
            resourceInputs["externalOpensearchLogsUserConfig"] = args ? args.externalOpensearchLogsUserConfig : undefined;
            resourceInputs["externalSchemaRegistryUserConfig"] = args ? args.externalSchemaRegistryUserConfig : undefined;
            resourceInputs["jolokiaUserConfig"] = args ? args.jolokiaUserConfig : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["prometheusUserConfig"] = args ? args.prometheusUserConfig : undefined;
            resourceInputs["rsyslogUserConfig"] = args ? args.rsyslogUserConfig : undefined;
            resourceInputs["signalfxUserConfig"] = args ? args.signalfxUserConfig : undefined;
            resourceInputs["endpointConfig"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServiceIntegrationEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceIntegrationEndpoint = ServiceIntegrationEndpoint;
/** @internal */
ServiceIntegrationEndpoint.__pulumiType = 'aiven:index/serviceIntegrationEndpoint:ServiceIntegrationEndpoint';
